nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -0,0 +1,410 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import json
+import os
+import pickle as pkl
+import shutil
+import tarfile
+import tempfile
+from collections import defaultdict
+from itertools import repeat
+from multiprocessing import Pool
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+from os import path
+import torch
+from omegaconf import DictConfig, OmegaConf, open_dict
+from pytorch_lightning.utilities import rank_zero_only
+from tqdm import tqdm
+from nemo.core.classes import Model
+from nemo.utils import config_utils, logging, model_utils
+from nemo.collections.asr.parts.mixins import DiarizationMixin
+from nemo.collections.asr.models.classification_models import EncDecClassificationModel
+from nemo.collections.asr.models.label_models import ExtractSpeakerEmbeddingsModel
+from nemo.collections.asr.parts.speaker_utils import get_score
+from nemo.collections.asr.parts.vad_utils import gen_overlap_seq, gen_seg_table, get_status, write_manifest
+from nemo.utils import logging
+from nemo.utils.exp_manager import NotFoundError
+
+try:
+    from torch.cuda.amp import autocast
+except ImportError:
+    from contextlib import contextmanager
+
+    @contextmanager
+    def autocast(enabled=None):
+        yield
+
+
+__all__ = ['ClusteringDiarizer']
+
+_MODEL_CONFIG_YAML = "model_config.yaml"
+_VAD_MODEL = "vad_model_.nemo"
+_SPEAKER_MODEL = "speaker_model.nemo"
+
+
+class ClusteringDiarizer(Model, DiarizationMixin):
```

Comment:
@okuchaiev @titu1994 Here are the changes that were suggested to make this "model" inherit from Model instead of ModelPT. Please take a look when you get a chance. I have tested saving and restoring functionalities and they work fine.